### ISSUE TITLE
Bug: false positive @typescript-eslint/no-unused-vars errors since update

### ISSUE THREAD

**Issue by plehnen** [typescript-eslint/typescript-eslint#7399]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

After updating ... 

```
"@typescript-eslint/eslint-plugin": "^5.62.0",
"@typescript-eslint/parser": "^5.62.0",
```

...to a newer version, running eslint shows many **@typescript-eslint/no-unused-vars** errors, which aren't errors:

i.e.

```
export const XY: InjectionKey<Ref<boolean | ((arg: string) => boolean)>> = Symbol('XY'); // 'arg' is defined but never used              @typescript-eslint/no-unused-vars

export type CustomFiltersType = Record<
  string,
  {
    label?: (value: any) => string; // 'value' is defined but never used  @typescript-eslint/no-unused-vars
  }

// in vue props:
    disabledDate: {
      type: Function as PropType<(date: string) => boolean>, // 'date' is defined but never used  @typescript-eslint/no-unused-vars
    },
>;
```

Maybe related to #7211, but what's written there didn't solve it.

### Reproduction Repository Link

https://github.com/plehnen/eslint-demo

### Repro Steps

1) clone the repo
2) `yarn install`
3) `yarn fix:lint`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `^6.2.1` |
| `@typescript-eslint/parser`            | `^6.2.1` |
| `@typescript-eslint/typescript-estree` | `^6.2.1` |
| `TypeScript`                           | `5.1.6` |
| `ESLint`                               | `8.46.0` |
| `node`                                 | `18.16.0` |

**Comment by bradzacher** [comment]
> but what's written there didn't solve it.

We'll that makes sense because the instructions in #7211 instructed you to add overrides for `.ts` files.

But you're seeing problems in `.vue` files...

It looks like the Vue stack doesn't currently support our v6 yet (eg https://github.com/vuejs/eslint-config-typescript/issues/57).

You will have to wait for them to update before moving to the major, unfortunately.
